pulsar-metadata/src/main/java/org/apache/pulsar/metadata/tableview/impl/MetadataStoreTableViewImpl.java

```diff
@@ -0,0 +1,339 @@
+        if (!isValidPath(path)) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return cache.get(path)
+                .thenAccept(valOpt -> {
+                    valOpt.ifPresent(val -> {
+                        String key = getKey(path);
+                        updateData(key, val);
+                        if (log.isDebugEnabled()) {
+                            log.debug("{} applying existing item key={} value={}",
+                                    name,
+                                    key,
+                                    val);
+                        }
+                        for (var listener : existingItemListeners) {
+                            try {
+                                listener.accept(key, val);
+                            } catch (Throwable e) {
+                                log.error("{} failed to listen existing item key:{}, val:{}", name, key, val,
+                                        e);
+                                throw e;
+                            }
+                        }
+                    });
+                });
+    }
+
+    private void fill() throws MetadataStoreException {
+        long started = System.currentTimeMillis();
+        log.info("{} start filling existing items under the pathPrefix:{}", name, pathPrefix);
+        ConcurrentLinkedDeque<String> q = new ConcurrentLinkedDeque<>();
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        q.add(pathPrefix);
+        LongAdder count = new LongAdder();
+        while (!q.isEmpty()) {
+            if (System.currentTimeMillis() - started > FILL_TIMEOUT_IN_MILLIS) {
+                String err = name + " failed to fill existing items in "
+                        + TimeUnit.MILLISECONDS.toSeconds(FILL_TIMEOUT_IN_MILLIS) + " secs. Filled count:"
+                        + count.sum();
+                log.error(err);
+                throw new MetadataStoreException(err);
+            }
+            int size = Math.min(MAX_CONCURRENT_METADATA_OPS_DURING_FILL, q.size());
+            for (int i = 0; i < size; i++) {
+                String path = q.poll();
+                futures.add(store.getChildren(path)
+                        .thenCompose(children -> {
+                            // The path is leaf
+                            if (children.isEmpty()) {
+                                count.increment();
+                                return handleExisting(path);
+                            } else {
+                                for (var child : children) {
+                                    q.add(path + "/" + child);
+                                }
+                                return CompletableFuture.completedFuture(null);
+                            }
+                        }));
+            }
+            try {
+                FutureUtil.waitForAll(futures).get(timeoutInMillis, TimeUnit.MILLISECONDS);
```

Comment:
Not a requested change. But it would be better to use a smaller value of left time and `timeoutInMillis` as the wait timeout.

```java
        final var deadline = System.currentTimeMillis() + FILL_TIMEOUT_IN_MILLIS;
        while (!q.isEmpty()) {
            // ...
            try {
                FutureUtil.waitForAll(futures).get(Math.min(timeoutInMillis, deadline - System.currentTimeMillis()),
                        TimeUnit.MILLISECONDS);
```

Reply:
updated.